dali/operators/reader/loader/filesystem.cc

```diff
@@ -22,14 +23,17 @@
 #include <string>
 #include <utility>
 #include <vector>
-
+#include "dali/core/call_at_exit.h"
 #include "dali/core/error_handling.h"
-#include "dali/operators/reader/loader/filesystem.h"
 #include "dali/operators/reader/loader/utils.h"
 
 namespace dali {
 namespace filesystem {
 
+inline bool starts_with(const std::string &str, const char *prefix) {
+  return str.rfind(prefix, 0) == 0;
```

Comment:
Why `rfind`? It would return false for `starts_with("blabla", "bla")` because rfind would return 3.
Also, why `find` at all? We want to compare the beginning, not look for a substring.
```suggestion
  return !prefix || !*prefix ||  // everything starts with an empty string
         (!str.empty() && strcmp(str.c_str(), prefix) == str.c_str());
```